### PR TITLE
Add practice attendance controls to app schedule

### LINKED ISSUE #1984: App: Practice — attendance taking
Part of the mobile app feature-parity backlog: practice attendance is recorded only on `drills.html` (`capabilities.ts` status: `stub`). Parents can already see attendance in the app.

**Scope (small unit)**
- Coach attendance screen for a practice session: mark present/absent/late per player, writing to `/practiceSessions/{sessionId}/attendance/{playerId}`

**Out of scope (separate issues)**
- Practice timeline, home packets

**Done when**
- Attendance taken in the app shows up in the web command center and parent dashboards

### PR DESCRIPTION
**Pull request by pauljsnider**
Closes #1984

Recovered from a partial issue-fixer run. The agent committed the fix branch successfully, so paulbot is opening the PR from the recorded commit.

### CODE REVIEW

File: apps/app/src/lib/scheduleService.ts [hunk 1]
```diff
@@ -2707,6 +2729,174 @@ function getPracticePacketSessionId(event: ParentScheduleEvent) {
...
+  }
+  if (!event.isDbGame) {
+    throw new Error('Practice attendance opens after this event is tracked in the schedule.');
+  }
+  if (!user?.uid) {
+    throw new Error('Sign in before managing practice attendance.');
+  }
+  if (!event.isTeamStaff) {
+    throw new Error('Only team coaches and admins can manage practice attendance.');
+  }
```
[the remark this answers is not in the document]
Comment (pauljsnider) [reply apps/app/src/lib/scheduleService.ts]: ### Remediation Update

Addressed this review feedback pass on PR #2007.

- Commit: `ea38a74776775038f143b1a2d060ab958dec7ca2`
- Validation: pass: 

{"success":true,"files_changed":["apps/app/src/lib/scheduleService.ts","apps/app/src/lib/scheduleService.test.ts","apps/app/src/pages/ScheduleEventDetail.tsx","apps/app/src/pages/ScheduleEventDetail.test.tsx"],"commit_message":"Fix practice attendance review findings","threads_addressed":["PRRT_kwDOQe-T586IWzD2","PRRT_kwDOQe-T586IWzD6"],"summary":"Aligned practice attendance access with the existing Firestore write boundary by requiring team owner/admin access in both the service guard and the app UI, so coach-only users no longer see controls that will always fail. Also prevented stale roster overwrites by optimistically updating local attendance state and disabling all attendance buttons while any save is in flight, with regression tests covering the authorization boundary and the overlapping-save behavior."}

File: apps/app/src/pages/ScheduleEventDetail.tsx [hunk 2]
```diff
@@ -2322,6 +2356,34 @@ function PracticePacketSection({ auth, event, childEvents }: { auth: AuthState;
...
+        : candidate
+    ));
+    setSavingAttendancePlayerId(player.playerId);
+    setAttendanceStatus(null);
+    setAttendanceError(null);
+    try {
+      const saved = await saveStaffPracticeAttendance(event, auth.user, {
+        ...attendance,
+        players: nextPlayers
+      });
```
[the remark this answers is not in the document]
Comment (pauljsnider) [reply apps/app/src/pages/ScheduleEventDetail.tsx]: ### Remediation Update

Addressed this review feedback pass on PR #2007.

- Commit: `ea38a74776775038f143b1a2d060ab958dec7ca2`
- Validation: pass: 

{"success":true,"files_changed":["apps/app/src/lib/scheduleService.ts","apps/app/src/lib/scheduleService.test.ts","apps/app/src/pages/ScheduleEventDetail.tsx","apps/app/src/pages/ScheduleEventDetail.test.tsx"],"commit_message":"Fix practice attendance review findings","threads_addressed":["PRRT_kwDOQe-T586IWzD2","PRRT_kwDOQe-T586IWzD6"],"summary":"Aligned practice attendance access with the existing Firestore write boundary by requiring team owner/admin access in both the service guard and the app UI, so coach-only users no longer see controls that will always fail. Also prevented stale roster overwrites by optimistically updating local attendance state and disabling all attendance buttons while any save is in flight, with regression tests covering the authorization boundary and the overlapping-save behavior."}

**Comment by pauljsnider** [comment]
### CI Fix Update

Addressed failing CI checks on PR #2007.

- Commit: `2a1a3551369b4a256415fcd51bef50f8e45f8887`
- Fixed checks: build-pages-bundle
- Validation: pass: 

{"success":true,"files_changed":["apps/app/src/pages/ScheduleEventDetail.test.tsx"],"commit_message":"fix:address-ci-failure: stabilize delayed attendance save test","checks_addressed":["build-pages-bundle [app-github-pages]"],"summary":"The CI failure came from a TypeScript control-flow issue in ScheduleEventDetail.test.tsx where a deferred resolver was initialized as null, so TypeScript narrowed the optional call target to never at build time. I changed the test to use a no-op function initializer and invoke it directly, preserving the async test behavior while making the file compile cleanly; this also prevents similar delayed-save test helpers from tripping TypeScript null narrowing."}
